File: lib/src/main/java/com/darklycoder/wifitool/lib/WiFiConfig.java
package com.darklycoder.wifitool.lib;

/**
 * WiFi配置
 */
public class WiFiConfig {

    //默认超时时间
    private static final long DEFAULT_TIME_OUT = 1000 * 15;

    public long timeOut = DEFAULT_TIME_OUT;

    private WiFiConfig() {
    }

    public static class Builder {
        private WiFiConfig config;

        public Builder() {
            config = new WiFiConfig();
        }

        public Builder setTimeOut(long time) {
            config.timeOut = time;
            return this;
        }

        public WiFiConfig build() {
            return config;
        }
    }

}

File: lib/src/main/java/com/darklycoder/wifitool/lib/WiFiConfig.kt
package com.darklycoder.wifitool.lib

/**
 * WiFi配置
 */
class WiFiConfig private constructor() {

    var timeOut = DEFAULT_TIME_OUT

    class Builder {
        private val config: WiFiConfig = WiFiConfig()

        fun setTimeOut(time: Long): Builder {
            config.timeOut = time
            return this
        }

        fun build(): WiFiConfig {
            return config
        }
    }

    companion object {

        // 默认超时时间
        private const val DEFAULT_TIME_OUT = (1000 * 15).toLong()
    }

}
